### DISCUSSION TITLE
how to change the network module in batch_eth_mnist

### DISCUSSION THREAD

**Discussion by MayWeHappy** [BindsNET/bindsnet#629] [Q&A]
I encountered an error when running the `network.run` function after replacing the network in `batch_eth_mnist` with `DiehlAndCook2015v2`. The error occurred in the `learning.py` file at line 399, specifically in the `_connection_update` method: `self.connection.w -= self.reduction(torch.bmm(source_s, target_x), dim=0)`.

The error message states that the output shape, [784, 100], does not match the broadcast shape, [32, 784, 100]. I tried using batch processing in other network training routines, such as SOM, but encountered the same issue.

Could you please advise on how to resolve this issue? Thank you very much.

**Comment by Hananel-Hazan** [comment]
The network `DiehlAndCook2015v2` is not designed to work with batch size. To adapt the network you need to modify the shape of the network input to match the batch size
